packages/presentation/lib/screen/home_page.dart

```diff
@@ -12,31 +11,52 @@ class HomePage extends StatefulWidget {
 }
 
 class _HomePageState extends State<HomePage> {
-  final PalindromeBloc bloc = PalindromeBloc();
+  final HomeBloc bloc = HomeBloc(PalindromeUseCaseImpl());
 
   @override
-  Widget build(BuildContext context) {
-    return Provider.value(
-      value: bloc,
-      child: Scaffold(
+  Widget build(BuildContext context) => StreamPlatformStackContent(
+        context: context,
+        dataStream: bloc.dataStream,
+        setText: bloc.setString,
+        checkPalindrome: bloc.checkPalindrome,
+      );
+}
+
+class StreamPlatformStackContent extends StatelessWidget {
+  final Stream<PalindromeData> dataStream;
+  final Function(String) setText;
+  final Function checkPalindrome;
+
+  const StreamPlatformStackContent({
+    Key? key,
+    required BuildContext context,
+    required this.dataStream,
+    required this.setText,
+    required this.checkPalindrome,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) => Scaffold(
         body: StreamBuilder(
-          stream: bloc.outputData,
+          stream: dataStream,
           builder:
-              (BuildContext context, AsyncSnapshot<PalindromeData?> snapshot) {
-            return _inputWidget(snapshot);
-          },
+              (BuildContext context, AsyncSnapshot<PalindromeData?> snapshot) =>
+                  _inputWidget(snapshot),
         ),
         floatingActionButtonLocation: FloatingActionButtonLocation.centerFloat,
         floatingActionButton: _showButtonWidget(),
-      ),
-    );
-  }
+      );
+
+  Widget _showButtonWidget() => FloatingActionButton(
+        onPressed: () => checkPalindrome(),
```

Comment:
if i'm not mistaken, u can use it in the following way:

`onPressed: checkPalindrome`

just fyi

Reply:
done